WikiWatershed/model-my-watershed
# Custom Weather: Text and Links

## Overview

Wires up the sample CSV file to be downloaded. It is now also used in testing, to store fewer CSV files in the repo, and to also validate our sample against our own rules.

Adds real text to the weather upload modal. May receive further tweaks from the client. Also makes clear that uploads need to be in centimeters and Celsius (the numbers themselves are not validated).

Also updates the scenario sidebar to report the correct number of years based on the custom weather file.

See commit messages for details.

Connects #3308 
Connects #3310

### Demo

Updated text:

![image](https://user-images.githubusercontent.com/1430060/82105140-bf574200-96e7-11ea-818a-f33dcd7511d2.png)

Updated weather information in sidebars:

![image](https://user-images.githubusercontent.com/1430060/82105146-caaa6d80-96e7-11ea-9879-4903d0b10b20.png)

![image](https://user-images.githubusercontent.com/1430060/82105157-d007b800-96e7-11ea-9b33-c6af911d5a1e.png)

![image](https://user-images.githubusercontent.com/1430060/82105164-d5fd9900-96e7-11ea-82ea-daf037f52728.png)

## Testing Instructions

* Check out this branch and `bundle --debug --vendor`
    - The vendor flag is required to get the new sample CSV in place as a static resource
* Create a MapShed project and add changes to it
* Open the Weather Data dialog
    - [x] Ensure it has updated text in it
    - [x] Ensure the links open in new tabs and work correctly
* Upload a custom weather file. Click "Save".
* Observe the Hydrology and Water Quality tabs on the left
    - [x] Ensure you see a line saying "Source: $YOUR_WEATHER_FILE"
    - [x] Ensure the title reflects the correct number of years of weather data
